src/io.jl

```diff
@@ -29,20 +29,24 @@ function load(fpath::String; gamma::Union{Nothing,Float64}=nothing, expand_palet
     close_png(fp)
     return out
 end
+
+maybe_lock(f, io::IO) = lock(f, io)
+# IOStream doesn't support locking...
+maybe_lock(f, io::IOStream) = f()
```

Comment:
According to the report in https://github.com/JuliaImages/Images.jl/issues/912#issuecomment-690183454

It looks like `IOBuffer` should also be included as a special case?

Reply:
Probably? I guess you could add that to #21